src/app/component/user/components/profile/profile-dashboard/profile-dashboard.component.ts

```diff
@@ -1,50 +1,68 @@
+import { HabitAssignInterface } from '../../../../../interface/habit/habit-assign.interface';
 
 @Component({
   selector: 'app-profile-dashboard',
   templateUrl: './profile-dashboard.component.html',
   styleUrls: ['./profile-dashboard.component.scss']
 })
-export class ProfileDashboardComponent implements OnInit {
-  public userId: number;
-  public menu: Array<HabitTab> = [
-    { id: 1, name: 'My habits', isActive: true },
-    { id: 2, name: 'My news', isActive: false },
-    { id: 3, name: 'My articles', isActive: false },
-  ];
-
-  public habitArray: Array<HabitItemModel> = [
-    {id: 1, dayCount: 5, title: 'Покупать местные продукты', describe: 'Mark as done', done: false, acquired: false},
-    {id: 2, dayCount: 12, title: 'Меньше пользоваться транспортом', describe: '5 days in a row. You’re good!', done: true, acquired: false},
-    {id: 3, dayCount: 5, title: 'Покупать местные продукты', describe: 'Mark as done', done: false, acquired: false},
-    {id: 4, dayCount: 12, title: 'Меньше пользоваться транспортом', describe: '2 days in a row. You’re good!', done: true, acquired: false},
-  ];
-
-  public habitAcquired: Array<HabitItemModel> = [
-    {id: 1, dayCount: 38, title: 'Брать кофе с собой в свою эко-чашку', describe: 'Great! You’ve got a habit.', done: true, acquired: true},
-    {id: 2, dayCount: 38, title: 'Брать кофе с собой в свою эко-чашку', describe: 'Great! You’ve got a habit.', done: true, acquired: true},
-    {id: 3, dayCount: 38, title: 'Брать кофе с собой в свою эко-чашку', describe: 'Great! You’ve got a habit.', done: true, acquired: true},
-  ];
-
-  constructor(private localStorageService: LocalStorageService) { }
+export class ProfileDashboardComponent implements OnInit, OnDestroy {
+  private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  loading = false;
+  habitsInProgress: Array<HabitAssignInterface> = [];
+  habitsAcquired: Array<HabitAssignInterface> = [];
+  public tabs = {
+    habits: true,
+    news: false,
+    articles: false
+  };
+  userId: number;
+
+  constructor(private localStorageService: LocalStorageService,
+              private habitService: HabitService,
+              private habitAssignService: HabitAssignService) { }
 
   ngOnInit() {
-    this.initUser();
+    this.executeRequests();
+    this.subscribeToLangChange();
+    this.getUserId();
+  }
+
+  getUserId() {
+    this.userId = this.localStorageService.getUserId();
+  }
+
+  subscribeToLangChange() {
+    this.localStorageService.languageBehaviourSubject
+      .pipe(takeUntil(this.destroyed$))
+      .subscribe(() => this.executeRequests());
+  }
+
+  private executeRequests() {
+    this.loading = true;
+    this.habitAssignService.getAssignedHabits()
+      .pipe(take(1))
+      .subscribe((response: Array<HabitAssignInterface>) => {
+        const sortedHabits = response
+          .sort((a, b) => (a.habit.id > b.habit.id) ? 1 : (b.habit.id > a.habit.id) ? -1 : 0);
```

Comment:
Move this array "sort" operation to a separate statement.

Reply:
fixed